source/connection-monitoring-and-pooling/tests/README.md

```diff
@@ -21,6 +21,74 @@ The following tests have not yet been automated, but MUST still be tested:
 5. When a check out attempt fails because connection set up throws an error, assert that a ConnectionCheckOutFailedEvent
     with reason="connectionError" is emitted.
 
+### Pending Response
+
+Due to the complexity of managing a proxy layer, the following qualifying tests should only be run for non-ssl,
+non-auth, and non-compression connections.
+
+#### Recover Partially Read Response
+
+This test verifies that if only part of a response was read before the timeout, the driver can drain the rest of the
+response and reuse the connection for the next operation.
+
+1. Connect to the proxy server with `maxPoolSize=1` and `direct=true`, subscribing to the following CMAP events:
+    - `PendingResponseStarted`
+    - `PendingResponseSucceeded`
+    - `PendingResponseFailed`
+    - `ConnectionClosed`
+2. Send a command (e.g. an insert) with a 200 millisecond timeout and the following `proxyTest` actions:
+    - `sendBytes`: We have 3 possible states here:
+        1. Message size was partially read: random value between 1 and 3 inclusive
+        2. Message size was read, body was not read at all: use 4
+        3. Message size was read, body read partially: random value between 5 and 100 inclusive
+    - `delayMS`: 400 ( to exceed the 200 ms timeout)
+    - `sendAll`: `true`
+3. Assert that the operation failed with timeout error.
+4. Issue another operation (e.g. another insert) and assert that it does not return an error.
+5. Verify that the following sequence of events was observed:
+    - `ConnectionPendingResponseStarted`
+    - `ConnectionPendingResponseSucceeded`
+6. Verify that NONE the following events was observed:
+    - `ConnectionPendingResponseFailed`
+    - `ConnectionClosed`
+
+#### Non-destructive Aliveness Check
+
+This test verifies that if a connection idles past the pending response window (3 seconds) after a partial header, the
+aliveness check does not attempt to discard bytes from the TCP stream.
```

Comment:
Aren't we supposed to close the connection if we receive no bytes in 3s?

Reply:
In scenario when connection is being sitting in the pool for some time longer then 3 seconds we do not want just to close the connection, because the response might be already buffered on the client side. We want to check if there are any new data arrived, if so we refresh the last read timestamp and performing draining as regular.

Reply:
I missed the sleep below.  thanks